NitroxServer/Serialization/World/WorldManager.cs

```diff
@@ -0,0 +1,149 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using NitroxModel.Helper;
+
+namespace NitroxServer.Serialization.World;
+
+public static class WorldManager
+{
+    public static readonly string SavesFolderDir = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), "Nitrox", "saves");
+
+    public static readonly string[] WorldFiles = { "BaseData.json", "EntityData.json", "PlayerData.json", "Version.json", "WorldData.json" };
+
+    private static readonly List<Listing> savesCache = new();
+
+    static WorldManager()
+    {
+        Directory.CreateDirectory(SavesFolderDir);
+    }
+
+    public static IEnumerable<Listing> GetSaves()
+    {
+        if (savesCache.Any())
+        {
+            return savesCache;
+        }
+
+        foreach (string folder in Directory.EnumerateDirectories(SavesFolderDir))
+        {
+            try
+            {
+                // Don't add the file to the list if it doesn't validate or contain a "server.cfg" file
+                string serverConfigFile = Path.Combine(folder, "server.cfg");
+                if (!ValidateSave(folder) || !File.Exists(serverConfigFile))
+                {
+                    continue;
+                }
+
+                Version version;
+                ServerConfig serverConfig = ServerConfig.Load(folder);
+
+                using (FileStream stream = new(Path.Combine(folder, "Version.json"), FileMode.Open, FileAccess.Read, FileShare.Read))
+                {
+                    version = new ServerJsonSerializer().Deserialize<SaveFileVersion>(stream)?.Version ?? NitroxEnvironment.Version;
+                }
+
+                // Change the paramaters here to define what save file versions are eligible for use/upgrade
+                bool IsValidVersion = true;
+                if (version < new Version(1, 5, 0, 1) || version > NitroxEnvironment.Version)
```

Comment:
What do you mean by that? I can upgrade 1.5.0.1 save files to 1.6.0.1, which I thought was intended. Otherwise, what is the point of the save file upgrade functionality?

That is there so that we can can change it whenever we make a major change to the way we have the save files, in which the save file upgrade functionality won't work